### ISSUE TITLE
SentenceTransformer fit function is not displaying loss function not for both training set and evaluation set

### ISSUE THREAD

**Issue by tkngoutham** [UKPLab/sentence-transformers#1209]
SentenceTransformer fit function is not explicitly computing loss function not for both training set and evaluation set

After going through the code of fit function of SentenceTransformer, loss function is being computed in torch modules. This is blocking us from evaluating if model is over fit or not.

Kindly consider this as new feature request to add feature to compute training loss and test loss explicitly.

**Comment by nreimers** [comment]
You need to pass an Evaluator to the fit function. Performance scores like accuracy, correlation, MRR are then printed if the logging level is set to info

**Comment by IvoryF** [comment]
Can't agree with it more.!
